shap/utils/_clustering.py

```diff
@@ -146,6 +146,8 @@ def hclust(X, y=None, linkage="single", metric="auto", random_state=0):
     if metric == "auto":
         if y is not None:
             metric = "xgboost_distances_r2"
+        else:
+            metric = "cosine"
```

Comment:
can we document this behaviour somewhere? Maybe in the (yet non-existent) docstring of the function

Reply:
Good idea, done.